src/encoded/commands/release_file.py

```diff
@@ -0,0 +1,491 @@
+            },
+            TISSUE: {
+                PC.SEQUENCING_READS: PC.PROTECTED,
+                PC.GERMLINE_VARIANT_CALLS: PC.PROTECTED,
+                PC.SOMATIC_VARIANT_CALLS: PC.OPEN,
+            },
+        }
+
+        if dataset in [
+            "colo829bl",
+            "colo829t",
+            "colo829blt_50to1",
+            "hapmap",
+            "hg002",
+            "hg00438",
+            "hg005",
+            "hg02257",
+            "hg02486",
+            "hg02622",
+        ]:
+            dataset_category = COLO829_HAPMAP
+        elif dataset in [
+            "lb_fibroblast",
+            "lb_ipsc_1",
+            "lb_ipsc_2",
+            "lb_ipsc_4",
+            "lb_ipsc_52",
+            "lb_ipsc_60",
+        ]:
+            dataset_category = IPSC
+        else:
+            self.print_error_and_exit(
+                f"Cannot get access_status from dataset {dataset}. Unknown dataset."
+            )
+
+        if len(file[PC.DATA_CATEGORY]) > 1:
+            self.add_warning(f"File has multiple data categories. Check access_status.")
+
+        data_category = file[PC.DATA_CATEGORY][0]
+
+        if data_category not in access_status_mapping[dataset_category]:
+            self.print_error_and_exit(
+                f"Cannot get access_status from data_category {data_category}. PLease add it to access_status_mapping."
+            )
+        return access_status_mapping[dataset_category][data_category]
+
+    def check_file_validity(self, file):
+        accession = file["accession"]
+
+        for prop in REQUIRED_FILE_PROPS:
+            if prop not in file:
+                self.print_error_and_exit(
+                    f"File {file[PC.ACCESSION]} does not have the required property `{prop}`."
+                )
+
+        if PC.FILE_SETS in file and len(file[PC.FILE_SETS]) > 0:
+            self.add_warning(
+                f"File {accession} already has an associated file set. It will be overwritten."
+            )
+
+        if file.get(PC.OUTPUT_STATUS) != PC.FINAL_OUTPUT:
+            self.add_warning(
+                f"File {accession} does not have {PC.OUTPUT_STATUS}='{PC.FINAL_OUTPUT}'."
+            )
+
+        if file.get(PC.STATUS) != PC.UPLOADED:
+            self.add_warning(
+                f"File {accession} has status `{file.get(PC.STATUS)}`. Expected `{PC.UPLOADED}`."
+            )
+
+    def print_error_and_exit(self, msg):
+        print(f"{bcolors.FAIL}ERROR: {msg} Exiting.{bcolors.ENDC}")
+        exit()
+
+    def add_warning(self, msg):
+        self.warnings.append(f"{bcolors.WARNING}WARNING:{bcolors.ENDC} {msg}")
+
+    def get_quality_metrics_zip_files(self, quality_metrics) -> List:
+        zip_files = []
+        for qm in quality_metrics:
+            url = qm.get("url", "")
+            # Parsing the URL here. Not ideal.
+            zip_accession = url.split("/")[-1]
+            zip_accession = zip_accession.split(".")[0]
```

Comment:
This is interesting; why isn't the zip file linked to the QualityMetric? Seems like it should be there.